deepchem/models/keras_model.py

```diff
@@ -432,8 +465,28 @@ def fit_generator(self,
         c(self, current_step)
       if self.tensorboard and should_log:
         self._log_scalar_to_tensorboard('loss', batch_loss, current_step)
-      if self.wandb and should_log:
-        wandb.log({'loss': batch_loss}, step=current_step)
+      if self.wandb_logger is not None:
+
+        #Get epoch number
+        dataset_size = len(self.wandb_logger.datasets[self.wandb_logger.train_dataset_name])
+        batches_per_epoch = math.ceil(dataset_size / self.batch_size)
+        epoch_num = current_step // batches_per_epoch
+
+        #Log loss and metrics at specified interval
+        if should_log:
+          all_metrics = self.wandb_logger.calculate_metrics(self)
+          data = dict({'train/loss': batch_loss,
+                       'epoch': epoch_num})
+          data.update(all_metrics)
+          self.wandb_logger.log_data(data, step=current_step)
```

Comment:
```
all_metrics = self.wandb_logger.calculate_metrics(self)
data = dict({'train/loss': batch_loss,
                       'epoch': epoch_num})
data.update(all_metrics)
self.wandb_logger.log_data(data, step=current_step)
```

This could all be done in WandbLogger right?

So we'd have something like 1 line that would do all the logging.

`self.wandb_logger.log(self, current_step)`
  
  And the decision whether to do validation can also be determined in the logger

Reply:
I've removed the calculate_metrics (running into a logging issue earlier).
Best to keep it as log_data and log_metrics or combine into a single log(...)?

The wandblogger takes in a list/dict of dataset, can it be assumed that if someone passes in the dataset to the logger, then they would want it to be logged?